type/__package_pip/man.rst

```diff
@@ -9,7 +9,15 @@ cdist-type__package_pip - Manage packages with pip
 DESCRIPTION
 -----------
 Pip is used in Python environments to install packages.
-It is also included in the python virtualenv environment.
+It is also included in the Python virtualenv environment.
+
+**NB:** This type requires that pip is already installed on the target host.
+Since pip is tightly coupled to a specific Python interpreter and it is common
+to have multiple interpreters (e.g. CPython 2.x, 3.x and/or PyPy) and
+venvs (virtual environments) installed at the same time, this type does not try
+to guess which pip to use.
+Most OSes provide a package to get a default pip installation, so it's usually
+sufficient to add ``__package python3-pip`` to your manifest.
```

Comment:
`python3-pip` yes, but let's add something like "Installable package might have additional dependencies (e.g. some `-dev` package) which have to be installed from distro repository before calling __package_pip"...

Reply:
What do you mean with "some `-dev` package"?
Pip packages that don't have a wheel for your platform and you need to compile native code directly on the target?

If so, I think we'd be going too far.
Some distros need `-dev` packages (or whatever the suffix is), some don't, on others you compile things from source anyway so you already have the headers.
And if you need to compile native code, you'll also need a compiler toolchain installed. That's usually done by installing `build-essential`/`build-base` or whatever, but then things aren't limited to C. Some people (looking at you, cryptography) want to bundle Rust code, so you need a Rust toolchain, too.

What I'm trying to say is, things are complicated and I think that compressing it down to a paragraph in the man page is not going to work.
Our users are admins, they should be able to handle it.

Reply:
We could add a sentence on `python3-wheel`, maybe.